### ISSUE TITLE
Disable specific rules for specific locators in axe-core scan

### ISSUE THREAD

**Issue by ling1726** [dequelabs/axe-core#4417]
Product: playwright

Expectation: Disable rules for a specific selector in an axe scan

Actual: Can either disable

* All rules for one locator and its descendants
* One rule for the entire scan

Motivation: Even if the scanned element has an invalid rule inside one of its subtrees. It's still worth running other rules for that its subtree. 

Let's take this example

```html
<div id="to-test">
  <button id="duplicate-id">
    <i class="send-icon"><i>
  </button>

  <button id="duplicate"><i class="delete-icon"></i></button>
</div>
```

There are two rules that are being violated here:
1. [button does not have accessible name](https://dequeuniversity.com/rules/axe/4.1/button-name)
2. [ids are not unique](https://dequeuniversity.com/rules/axe/4.3/duplicate-id#:~:text=Duplicate%20IDs%20are%20common%20validation,be%20sure%20each%20is%20unique.)

If I test the entire UI markup here, there's no way for me ignore a specific rule for one of the problematic buttons (if they happen to be an exception). In fact, it's possible to mistakenly exclude one of the problematic buttons or rules and get a less accurate result from the scan.

<pre><code>
axe-core version: 4.9.0
@axe-core/playwright: 4.9.0

- Node version: 20
- Platform:  Mac, Linux, Windows
</code></pre>

**Comment by dbjorge** [comment]
Today, the easiest way to achieve this is by post-processing the results from axe-core before writing your test assertions. [Playwright's Accessibility Testing docs](https://playwright.dev/docs/accessibility-testing#using-snapshots-to-allow-specific-known-issues) has a section suggesting one option for how to do that.

Longer-term, we do agree that this is a feature that would make sense to add to axe-core. We don't currently have engineering cycles to prioritize for it, but it's something we'd be happy to accept a contribution for.

We did discuss some different options for it and we're leaning towards something along these lines:

- Using a similar "inline suppression" concept to what other dev tools use for one-off issue suppression (eg, `// prettier-ignore` or `//  eslint-disable-next-line rule-name`)
- ...by adding support for putting an inline suppression attribute on elements, something along the lines of `<div data-axe-skip="color-contrast">`.
- We might want different syntax to represent "skip for just this element" vs "skip for this element and its descendants".
- We don't think we'd want to completely remove those results from the final result object - we think it'd probably make more sense to instead list them out in a new `skipped` array on the result objects, which would be a sibling of the existing `violations`/`incomplete`/etc arrays.

We also considered adding properties to the axe.run `Options` or `Context` types to represent elements that should be skipped, but we think we'd prefer the inline option, at least to start with; we think it would be less complex to make that work consistently across all the languages and products that we maintain axe integrations for.

**Comment by lukpsaxo** [comment]
this is possible today by passing a selector and rule and using the not( selector to exclude specific elements.

however it seems axe-core-npm don't want to support this:

https://github.com/dequelabs/axe-core-npm/issues/975#issuecomment-1887596887
